### PR TITLE
Feat: Increase one-shot responses, reduce time & cost by adding conditional logic to write_to_file

### PR DESCRIPTION
**Pull request by lc0rp**
### Background
write_to_file is one of the most used commands.
Because it doesn't support any conditional logic, Auto-GPT often has to combine it with another command, to aid its decision-making.

This PR adds support for "write_if_exists", "skip_if_exists" which often results in one less cycle for Auto-GPT.

### Changes
- Added an "if_exists" parameter which can be either 'overwrite', 'prepend', 'append', 'skip' or 'fail'
- CI (mypy) required some other changes to related files.

### Documentation
No changes are needed.

### Test Plan
Existing tests.

### PR Quality Checklist
- [X] My pull request is atomic and focuses on a single change.
- [X] I have thoroughly tested my changes with multiple different prompts.
- [X] I have considered potential risks and mitigations for my changes.
- [X] I have documented my changes clearly and comprehensively.
- [X] I have not snuck in any "extra" small tweaks changes. 
- [x] I have run the following commands against my code to ensure it passes our linters:
    ```shell
    black .
    isort .
    mypy
    autoflake --remove-all-unused-imports --recursive --ignore-init-module-imports --ignore-pass-after-docstring autogpt tests --in-place
    ```

### CODE REVIEW

**Review by JukoiDev — APPROVED** [review]
lgtm!

File: autogpt/commands/file_operations.py [hunk 1]
```diff
@@ -204,32 +213,63 @@ def ingest_file(
             "description": "The text to write to the file",
             "required": True,
         },
+        "if_exists": {
+            "type": "string",
```
Comment (Pwuts) [inline autogpt/commands/file_operations.py]: Best to implement Enum parameter type support for this

**Review by Pwuts — CHANGES_REQUESTED** [review]

File: autogpt/commands/file_operations.py (hunk 1 above)
Comment (angelisab1) [reply autogpt/commands/file_operations.py]: Reply

Comment (angelisab1) [inline autogpt/commands/file_operations.py]: autogpt/commands/file_operations.py

Comment (angelisab1) [reply autogpt/commands/file_operations.py]: if_exists": { "type": "string",

Comment (angelisab1) [reply autogpt/commands/file_operations.py]: @zamaniego1

Comment (angelisab1) [reply autogpt/commands/file_operations.py]: autogpt/commands/file_operations.py

File: autogpt/commands/file_operations.py [hunk 2]
```diff
@@ -204,32 +213,63 @@ def ingest_file(
...
+
     try:
-        directory = os.path.dirname(filename)
-        os.makedirs(directory, exist_ok=True)
-        with open(filename, "w", encoding="utf-8") as f:
-            f.write(text)
-        log_operation("write", filename, agent, checksum)
+        result = ACTIONS.get(
+            if_exists, lambda p, _txt: "Error: Invalid value for 'if_exists'."
+        )(path, text)
```
Comment (angelisab1) [inline autogpt/commands/file_operations.py]: autogpt/commands/file_operations.py

**Review by angelisab1 — APPROVED** [review]
tests/unit/test_file_operations.py

**Comment by angelisab1** [comment]
> lgtm!

me ayudas

**Pull request closed without merging**